service/src/main/java/org/cbioportal/service/impl/GenePanelServiceImpl.java

```diff
@@ -23,11 +25,13 @@
 public class GenePanelServiceImpl implements GenePanelService {
     @Autowired
     private GenePanelRepository genePanelRepository;
+    
+    private final Integer maxCasesCountToIncludeInQuery = 30000;
```

Comment:
@kalletlak this number works now but it might be nice to get some idea of how we should update it based on max cohort size. I imagine if you have 1M samples in a cohort it might be better to increase the threshold? So it's somewhat study size dependent, correct?